core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -389,6 +389,9 @@ class BrokerServer(
       // a potentially lengthy recovery-from-unclean-shutdown operation here, if required.
       metadataListener.startPublishing(metadataPublisher).get()
 
+      // Log static broker configurations.
+      new KafkaConfig(config.originals(), true)
```

Comment:
A couple of questions:
1. Where are we logging this for the zk case?
2. Can we expose a method in `KafkaConfig` to write the configs to a logger or something? Seems a bit cleaner than this approach.

Reply:
> 1. Where are we logging this for the zk case?
It's logged from DynamicBrokerConfig.initialize (see above stack trace)

> 2. Can we expose a method in KafkaConfig to write the configs to a logger or something? Seems a bit cleaner than this approach.
Can we do this after 3.0? Due to the inheritance-based design here, there isn't a simple way to expose this without changing the base class. The comment does explain what the line does...

Reply:
Can we file a JIRA?

Reply:
I also learned that the ZK path logs dynamic configs too now. So, we can have a JIRA about logging dynamic configs for KRaft and mention there the potential clean-up.

Reply:
Filed https://issues.apache.org/jira/browse/KAFKA-13105